XCUITests/AuthenticationTest.swift

```diff
@@ -0,0 +1,496 @@
+    }
+    
+    override func tearDown() {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+        super.tearDown()
+    }
+
+    private func openAuthenticationManager() {
+        let app = XCUIApplication()
+        app.buttons["TabToolbar.menuButton"].tap()
+        app.collectionViews.cells["SettingsMenuItem"].tap()
+        
+        let appsettingstableviewcontrollerTableviewTable = app.tables["AppSettingsTableViewController.tableView"]
+        waitforExistence(appsettingstableviewcontrollerTableviewTable.cells["TouchIDPasscode"])
+        appsettingstableviewcontrollerTableviewTable.cells["TouchIDPasscode"].tap()
+        waitforExistence(app.staticTexts["Passcode"])
+    }
+
+    private func closeAuthenticationManager() {
+        let app = XCUIApplication()
+
+        app.navigationBars["Passcode"].buttons["Settings"].tap()
+        app.navigationBars["Settings"].buttons["AppSettingsTableViewController.navigationItem.leftBarButtonItem"].tap()
+    }
+    
+    private func disablePasscode(passCode: String) {
+        let app = XCUIApplication()
+        
+        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode Off"].tap()
+        waitforExistence(app.staticTexts["Enter passcode"])
+        app.keys[passCode[0]].tap()
+        app.keys[passCode[1]].tap()
+        app.keys[passCode[2]].tap()
+        app.keys[passCode[3]].tap()
+        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"])
+    }
+    
+    private func enablePasscode(passCode: String) {
+        let app = XCUIApplication()
+        
+        app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode On"].tap()
+        waitforExistence(app.staticTexts["Enter a passcode"])
+        app.keys[passCode[0]].tap()
+        app.keys[passCode[1]].tap()
+        app.keys[passCode[2]].tap()
+        app.keys[passCode[3]].tap()
+        waitforExistence(app.staticTexts["Re-enter passcode"])
+        app.keys[passCode[0]].tap()
+        app.keys[passCode[1]].tap()
+        app.keys[passCode[2]].tap()
+        app.keys[passCode[3]].tap()
+        waitforExistence(app.tables["AuthenticationManager.settingsTableView"].staticTexts["Turn Passcode Off"])
+     }
+
+    // Sets the passcode and interval (set to immediately)
+    func testTurnOnOff() {
+        let app = XCUIApplication()
+
+        openAuthenticationManager()
+        enablePasscode("1337")
+        XCTAssertTrue(app.staticTexts["Immediately"].exists)
+        
+        disablePasscode("1337")
+        closeAuthenticationManager()
+    }
+    
+    func testChangePassCode() {
+        let app = XCUIApplication()
+        
+        openAuthenticationManager()
+        enablePasscode("1337")
+        app.staticTexts["Change Passcode"].tap()
+        waitforExistence(app.staticTexts["Enter passcode"])
+        app.keys["1"].tap()
```

Comment:
Yup, this is a very rough copy, will need to do some polishing tomorrow. I put a comment regarding the bug i found - can i still submit when the test is failing? (Assuming the bug is valid)